recipe/build.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -e # Abort on error.
+
+export PING_SLEEP=30s
+export WORKDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+export BUILD_OUTPUT=$WORKDIR/build.out
+
+touch $BUILD_OUTPUT
+
+dump_output() {
+   echo Tailing the last 500 lines of output:
+   tail -500 $BUILD_OUTPUT
+}
+error_handler() {
+  echo ERROR: An error was encountered with the build.
+  dump_output
+  exit 1
+}
+
+# If an error occurs, run our error handler to output a tail of the build.
+trap 'error_handler' ERR
+
+# Set up a repeating loop to send some output to Travis.
+bash -c "while true; do echo \$(date) - building ...; sleep $PING_SLEEP; done" &
+PING_LOOP_PID=$!
+
+## START BUILD
```

Comment:
@Korijn all this boilerplate is just to keep pinging, so Travis-CI does not quit due to lack of response or the 4 MB log limit while also saving the log in the `$BUILD_OUTPUT`.

We are not compiling on OS X for now but I wanted to put this here anyways.